### PR TITLE
adding numa install for linux

### PR DESCRIPTION
**Pull request by nitinsingh130519**
This is to add numa installation step for linux

Tested and below is the screenshot

image

### CODE REVIEW

File: perf/benchmark_setup_linux.sh [hunk 1]
```diff
@@ -176,7 +182,7 @@ function set_thp_madvise() {
 # Parse and handle command-line options
 function parse_arguments() {
     if [[ "$#" -eq 0 ]]; then
-        # Default: run all functions if no options provided
+    # Default: run all functions if no options provided
```
Comment (karianna) [inline perf/benchmark_setup_linux.sh]: looks like whitespace is out.

File: perf/benchmark_setup_linux.sh [hunk 2]
```diff
@@ -19,10 +19,16 @@ function show_help() {
...
     echo "=========================================================="
     echo "Checking NUMA settings with numactl"
+
+    # Check if numactl is installed, install it if not
     if ! command -v numactl &> /dev/null; then
-        echo "WARNING: numactl is not installed or not found. Skipping NUMA check."
-        return 0  # Skip without error
+        echo "'numactl' is not installed. Installing it now..."
+        if  ! sudo apt install  numactl; then
+            echo "ERROR: Failed to install 'numactl'. Exiting."
```
Comment (karianna) [inline perf/benchmark_setup_linux.sh:27]: probably worth echoing out what the result of the attempt was.

Comment (nitinsingh130519) [reply perf/benchmark_setup_linux.sh:27]: @karianna doesnt this 
image
logs the required information ? or am I missing something

**Comment by nitinsingh130519** [comment]
Output for windows 
image

File: perf/benchmark_setup_linux.sh (hunk 2 above)
Comment (karianna) [reply perf/benchmark_setup_linux.sh:27]: I was thinking the error case, so echoing why the install failed.  On second thoughts though, the user can just rerun that manually and see.

File: perf/benchmark_setup_windows.bat [hunk 3]
```diff
@@ -0,0 +1,112 @@
+@echo off
+
+REM ===========================================================
+REM Script for Windows System Configuration
+REM - NUMA Node Information
+REM - Power Mode Configuration
+REM Author: Nitin
```
Comment (karianna) [inline perf/benchmark_setup_windows.bat]: Can remove the author here (will pick up off commits)

File: perf/benchmark_setup_windows.bat [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+REM Function to set Power Mode to High or Ultimate Performance
+:set_performance
+echo ===========================================================
+echo Checking Available Power Modes...
+echo ===========================================================
+powercfg -list
+
+REM Check if Ultimate Performance mode exists
+set ULTIMATE_GUID=e9a42b02-d5df-448d-aa00-03f14749eb61
```
Comment (karianna) [inline perf/benchmark_setup_windows.bat:66]: Is this GUID a const across all systems?

Comment (nitinsingh130519) [reply perf/benchmark_setup_windows.bat:66]: yes

**Review by karianna — APPROVED** [review]

**Pull request merged**